### PR TITLE
Script to fetch all Issues/PRs and store them in a CSV/JSON File #932

### LINKED ISSUE #932: Script to fetch all Issues/PRs and store them in a CSV/JSON File
# Description

Create a Script to fetch all Issues/PRs and store them in a CSV/JSON File. It should ideally work as a handy CLI tool that can allow the maintainers to capture the data trends of the repository. For the Issues, capture the following data points: 

- Issue Creator
- Issue Assignee 
- Corresponding PR (if any)
- Issue Open/Close Date 
- People who commented on the Issue (Numbers and their Username) 
- Issue Content

For the PRs capture the following data points: 

- PR Creator
- Merged/Closed 
- PR Open/Close/Merge Date 
- Reviewers (Numbers and their Username) 
- Who merged the PR? 
- PR Content
- Files changed

## Language

- [x] Go
- [x] Javascript
- [x] Python

### PR DESCRIPTION
**Pull request by Jaideep07**
# Description

The Script fetchs all Issues/PRs and stores them in a JSON File. It ideally works as a handy CLI tool that can allow the maintainers to capture the data trends of the repository.

Fixes  #932 

Replace `932` with the issue number which is fixed in this PR

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Jaideep07** [comment]
I think the linter is failing because of .json files, otherwise, i have linted the script with flake8

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
Remove all the images from the codebase, use an image hosting website, Please read guidelines.

Rename the directory, the name is too long. 

Read the styling guideline and reformat your README, you can even use [this](https://github.com/HarshCasper/Rotten-Scripts/runs/2286635868?check_suite_focus=true)

**Comment by Jaideep07** [comment]
I tried to name it as small as possible and also tried to keep it meaning full, pls do suggest a better and shorter name if possible 😃

File: Python/Github_Issues_PRs_Scrapper/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+PyGithub
```
Comment (vybhav72954) [inline Python/Github_Issues_PRs_Scrapper/requirements.txt:1]: Please include the version number here. Use `pip freeze` for that

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: Python/Github_Issues_PRs_Scrapper/requirements.txt (hunk 1 above)
Comment (Jaideep07) [reply Python/Github_Issues_PRs_Scrapper/requirements.txt]: Done!

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Script to fetch all Issues/PRs and store them in a CSV/JSON File
+
+[![forthebadge made-with-python](http://ForTheBadge.com/images/badges/made-with-python.svg)](https://www.python.org/)
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/README.md:3]: No need of this

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+## The Script fetchs all Issues/PRs and stores them in a JSON File, It ideally works as a handy CLI tool that can allow the maintainers to capture the data trends of the repository
+
+### For the Issues, it capture the following data points
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/README.md:7]: No need for heading tags

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+- Who merged the PR?
+- PR Content
+- Files changed
+
+## Dependencies
+
+ 1. python3
+ 2. PyGithub
+ 3. json
+ 4. sys
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/README.md:31]: Use bullet points here

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Setup instructions
+
+- Install the dependencies using ```pip3 install -r requiremnts.txt```
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/README.md:35]: Typo in `requirements`

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+## Output
+
+- Terminal
+[![terminal-OUT.png](https://i.postimg.cc/5NLFRZHb/terminal-OUT.png)](https://postimg.cc/jLq5W3QF)
+- AllIssuesInfo.json
+[![issuesSS.png](https://i.postimg.cc/7PX3VvTc/issuesSS.png)](https://postimg.cc/JtDHr2xc)
+- AllPullRequestsInfo.json
+[![prSS.png](https://i.postimg.cc/Qx9FWvgc/prSS.png)](https://postimg.cc/svzjk6N2)
+
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/README.md:49]: Just a terminal output would suffice I guess. We don't need the JSON screenshots. Or else a GIF combining all would be even better.

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+## Author(s)
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/README.md:50]: Only `Author`

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+Made by Jaideep Reddy Kotla
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/README.md:52]: Add a hyperlink for your profile (GitHub/LinkedIn)

File: Python/Github_Issues_PRs_Scrapper/master_fetch_script.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    with open(str(fileName) + ".json", "w") as file:
+        json.dump(data, file, indent=6)
+        file.close()
+    print("Successfully created " + str(fileName) + ".json")
+
+
+if __name__ == "__main__":
+
+    # Setup your github token before running the script
+    g = Github("***********IGHRh*****************hQ")
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/master_fetch_script.py:76]: Can you create a `.env` file and store the token there? Use a module like `decouple` to import them in your script and use them on the fly.

Also don't push `.env` on the PR. Instead push a `.env.example` highlighting the token that the user needs to keep.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/Github_Issues_PRs_Scrapper/.env.example [hunk 10]
```diff
@@ -0,0 +1 @@
+GITHUBTOKEN=duieaX12nui@8jivdrlmkvs01nkjvseior1kvmos12345
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/.env.example:1]: Is this your actual token? If yes, kindly remove it from the commits and rebase.

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+#### For the PRs capture the following data points
+
+- PR Creator
+- Merged/Closed
+- PR Open/Close/Merge Date
+- Reviewers (Numbers and their Username)
+- Who merged the PR?
+- PR Content
+- Files changed
```
Comment (HarshCasper) [inline Python/Github_Issues_PRs_Scrapper/README.md:18]: Can you please follow a standard template that we have on the home directory?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/Github_Issues_PRs_Scrapper/.env.example (hunk 10 above)
Comment (Jaideep07) [reply Python/Github_Issues_PRs_Scrapper/.env.example:1]: Nope😁, it's not my token

Comment (vybhav72954) [reply Python/Github_Issues_PRs_Scrapper/.env.example:1]: Still its not supposed to look like this. See the token in this [PR](https://github.com/HarshCasper/Rotten-Scripts/pull/896)

File: Python/Github_Issues_PRs_Scrapper/README.md (hunk 11 above)
Comment (Jaideep07) [reply Python/Github_Issues_PRs_Scrapper/README.md:18]: @HarshCasper, you mean the rotten scripts readme, like heading, an introduction followed by the dependencies ..... etc ? will this order be okay?

Comment (vybhav72954) [reply Python/Github_Issues_PRs_Scrapper/README.md:18]: Follow [this](https://github.com/HarshCasper/Rotten-Scripts/blob/master/TEMPLATE_README.md)

Comment (Jaideep07) [reply Python/Github_Issues_PRs_Scrapper/README.md:18]: oh okay 👍

File: Python/Github_Issues_PRs_Scrapper/README.md [hunk 12]
```diff
@@ -1,4 +1,8 @@
-### For the Issues, it capture the following data points
+# Script to scrape all the issues and pull requests of GitHub repo
```
Comment (vybhav72954) [inline Python/Github_Issues_PRs_Scrapper/README.md:1]: The name is too long, Keep the title as short as possible.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**